lib/mock/mocks.go

```diff
@@ -4,67 +4,44 @@ import (
 	"context"
 	"io/ioutil"
 
+	"github.com/odpf/columbus/discovery"
 	"github.com/odpf/columbus/lineage"
-	"github.com/odpf/columbus/models"
+	"github.com/odpf/columbus/record"
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/mock"
 )
 
```

Comment:
no particular reason really.
most of the mocks were already there and I decided to not change the legacy that much to make the restructure smoother and quicker by not introducing more new things.

But If it improves our development experience, im more than happy to use that lib.
And since it is outside of the scope of this PR, I created this issue to track your suggestion.
https://github.com/odpf/columbus/issues/26 🙏